### ISSUE TITLE
Invocation parsing bug

### ISSUE THREAD

**Issue by jonathandturner** [nushell/nushell#2921]
**Describe the bug**
If you try to use ranges from inside of an invocation, it looks like sometimes it'll read the wrong range and see that as the variable:

```
echo $(echo 1..$foo | each { echo $it | describe })
error: Variable not in scope
  ┌─ shell:4:6
  │
4 │ echo $(echo 1..$foo | each { echo $it | describe })
  │      ^^^^^^^^ unknown variable
```

**To Reproduce**
1. See above

**Expected behavior**
This should parse visually as expected, where everything inside of the invocation parses the same way as it would if it was on a line by itself.

Latest main

**Comment by fdncred** [comment]
Trying to get this one working and i'm getting `variable not in scope` - may be related
```
echo 0..8 | each {
    let x = $it
    echo 30..37 | each {
        let i = $it
        echo 40..47 | each {
            let a = $it
            let color = $(build-string $x ';' $i ';' $a 'm')
            echo [$(ansi -e $color) $color $(ansi -e '0;37;40m')]
        }
    }
}

error: Variable not in scope
   ┌─ shell:18:40
   │
18 │             let color = $(build-string $x ';' $i ';' $a 'm')
   │                                        ^^ unknown variable

error: External command failed
   ┌─ shell:25:1
   │
25 │ nu color_tables.nu
   │ ^^ command failed
```
could be related to #2901 as well

**Comment by fdncred** [comment]
Strangely enough this works, and is very similar to the above script but with `str collect`s.
```
echo 0..8 | each {
    let x = $it
    let row = $(echo 30..37 | each {
        let i = $it
        let row = $(echo 40..47 | each {
            let a = $it
            let color = $(build-string $x ';' $i ';' $a 'm')
            echo [$(ansi -e $color) $color $(ansi reset) '  ']
        } | str collect)
        echo [$row $(char newline)] | str collect
    } | str collect)
    echo [$row $(char newline)] | str collect
} | str collect
```

**Comment by jonathandturner** [comment]
This should be fixed by #2945
